### ISSUE TITLE
flock system call  on a dir  return bad file descriptor  when using ego run 

### ISSUE THREAD

**Issue by ParkerXie** [edgelesssys/ego#113]
Hi，When run my program using ego run ,I encounter a problem:  if the code  try to use flock system call on a dir that return the error " bad file descriptor", But it work fine  when  use flock system call on file.
sample code:

```golang

package main

import (
	"fmt"
	"os"
	"golang.org/x/sys/unix"
)

func LockFile(fileName string) error {
	f, err := os.Create(fileName)
	if err != nil {
	   return err
	}
	return unix.Flock(int(f.Fd()), unix.LOCK_EX|unix.LOCK_NB)
}

func LockDir(dir string) error {
	if err := os.MkdirAll(dir, os.ModePerm); err != nil {
	   return  err
	}
	f, err := os.Open(dir)
	if err != nil {
	    return err
	}
	return unix.Flock(int(f.Fd()), unix.LOCK_EX|unix.LOCK_NB)
}

func main() {
	err := LockFile("testfile")
	if err != nil {
		fmt.Printf("testfile lock error: %v",err)
	}
	err = LockDir("testdir")
	if err != nil {
		fmt.Printf("testdir lock error: %v",err)
	}
}
```

**Comment by thomasten** [comment]
Thanks for reporting. We can reproduce the issue and will investigate the cause.

**Comment by thomasten** [comment]
fixed in v0.4.1

**Comment by ParkerXie** [comment]
Thanks a lot
